Missing dependency NoClassDefFoundError: com/codahale/metrics/MetricRegistry

<!--
- Please follow the issue template below for bug reports and feature requests.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug or feature request, if an error is being thrown a stack trace helps -->
JHipster-registry fails to start with docker-compose configuration `jhipster-registry.yml`

##### **Motivation for or Use Case**

<!-- Explain why this is a bug or a new feature for you -->
JHipster-registry seems to be failing in certain situations

##### **Reproduce the error**
my `jhipster-registry.yml` file
```
version: '2'
services:
  jhipster-registry:
    image: jhipster/jhipster-registry:v5.0.0
    volumes:
      - ./central-server-config:/central-config
    environment:
      - _JAVA_OPTIONS=-Xmx512m -Xms256m
      - SPRING_PROFILES_ACTIVE=dev,uaa
      - SPRING_SECURITY_USER_PASSWORD=admin
      - JHIPSTER_REGISTRY_PASSWORD=admin
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_TYPE=native
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_SEARCH_LOCATIONS=file:./central-config
      - JHIPSTER_LOGGING_LOGSTASH_ENABLED=true
      - JHIPSTER_LOGGING_LOGSTASH_HOST=jhipster-logstash
      - JHIPSTER_METRICS_LOGS_ENABLED=false
      - JHIPSTER_METRICS_LOGS_REPORTFREQUENCY=60
      - JHIPSTER_SECURITY_CLIENT_AUTHORIZATION_TOKEN_SERVICE_ID=uaa
    ports:
      - 8761:8761
```

then start the docker server with 'docker-compose -f hipster-registry.yml up' and I get the following error

```
java.lang.IllegalStateException: Error processing condition on org.springframework.boot.actuate.autoconfigure.metrics.MetricsAutoConfiguration.micrometerClock
   at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:64)
   at org.springframework.context.annotation.ConditionEvaluator.shouldSkip(ConditionEvaluator.java:108)

...

Caused by: java.lang.IllegalStateException: Failed to introspect Class [io.github.jhipster.config.metric.JHipsterLoggingMetricsExportConfiguration] from ClassLoader [org.springframework.boot.loader.LaunchedURLClassLoader@254989ff]
   at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:507)
   at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:404)
   at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:389)
   at org.springframework.util.ReflectionUtils.getUniqueDeclaredMethods(ReflectionUtils.java:447)
   at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1688)

...

jhipster-registry_1  | Caused by: java.lang.NoClassDefFoundError: com/codahale/metrics/MetricRegistry
   at java.lang.Class.getDeclaredMethods0(Native Method)
   at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
   at java.lang.Class.getDeclaredMethods(Class.java:1975)
   at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:489)
   	... 45 common frames omitted
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.MetricRegistry
   at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
   at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
   at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:93)
   at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
   	... 49 common frames omitted
```

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**
Add the missing dependency ??

it seems that commenting those two lines prevent it to fail ...
```
#     - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_TYPE=native
#     - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_SEARCH_LOCATIONS=file:./central-config
```
<!-- For bug reports, if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Registry Version(s)**
v5.0.0
<!--
Which version of JHipster Registry are you using, is it a regression?
-->

##### **Browsers and Operating System**
docker version 18.09.2 on macOS
<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [X] Checking this box is mandatory (this is just to show you read everything)

I generated a microservices architecture with JHipster 6.0.1 today and indeed everything fails to work because of this issue.

```
Caused by: java.lang.NoClassDefFoundError: com/codahale/metrics/MetricRegistry
at java.lang.Class.getDeclaredMethods0(Native Method)
at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
at java.lang.Class.getDeclaredMethods(Class.java:1975)
at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:489)
... 27 common frames omitted
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.MetricRegistry
at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:93)
at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
```

This is a major issue IMO.

--- a/pom.xml
+++ b/pom.xml
@@ -318,6 +318,11 @@
             <groupId>io.micrometer</groupId>
             <artifactId>micrometer-registry-prometheus</artifactId>
         </dependency>
+        <!-- Needed to enable log reporting to logstash through Dropwizard console reporter -->
+        <dependency>
+            <groupId>io.dropwizard.metrics</groupId>
+            <artifactId>metrics-core</artifactId>
+        </dependency>
         <!-- Java 11+ support -->
         <dependency>
             <groupId>org.glassfish.jaxb</groupId>

